### DISCUSSION TITLE
How to search a username?

### DISCUSSION THREAD

**Discussion by H1Mohamed** [pocketbase/pocketbase#3677] [Q&A] [answered]
I want to search the pocketbase users collection for a specific username,if it's found it will give me the record id,if not it will return a 404 not found error
Is it possible???
HOW CAN I DO THAT???

**Comment by ganigeorgiev** [comment, marked as answer]
Use a filter like `username="something"`. Here is an example with the JS SDK:

```js
const record = await pb.collection("users").getFirstListItem(
  // the same as `username="${yourUsername}"`
  pb.filter("username={:username}", { username: "something" })
)
```

**Reply by ganigeorgiev** [reply]
_Note that the above requires the client to satisfy the collection's List API rule._
